### PR TITLE
feat: Wormhole Bridge adapter

### PR DESCRIPTION
**Pull request by sendra**
Adds the Wormhole bridge adapter

### CODE REVIEW

File: src/contracts/adapters/wormhole/WormholeAdapter.sol [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    );
+
+    _registerReceivedMessage(payload, originChainId);
+  }
+
+  /// @inheritdoc IBaseAdapter
+  function nativeToInfraChainId(
+    uint256 nativeChainId
+  ) public pure virtual override returns (uint256) {
+    if (nativeChainId == uint16(2)) {
```
Comment (kyzia551) [inline src/contracts/adapters/wormhole/WormholeAdapter.sol]: Why do you cast to uint16 here and on the further conditions?

File: src/contracts/adapters/wormhole/WormholeAdapter.sol [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+      return 0;
+    }
+  }
+
+  /// @inheritdoc IBaseAdapter
+  function infraToNativeChainId(
+    uint256 infraChainId
+  ) public pure virtual override returns (uint256) {
+    if (infraChainId == ChainIds.ETHEREUM) {
+      return uint16(2);
```
Comment (kyzia551) [inline src/contracts/adapters/wormhole/WormholeAdapter.sol]: Why do you cast to uint16 here and on the further conditions?

File: src/contracts/adapters/wormhole/WormholeAdapter.sol [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    address srcAddress = address(uint160(uint256(sourceAddress)));
+
+    uint256 originChainId = nativeToInfraChainId(sourceChain);
+
+    require(
+      _trustedRemotes[originChainId] == srcAddress && srcAddress != address(0),
+      Errors.REMOTE_NOT_TRUSTED
+    );
+
+    _registerReceivedMessage(payload, originChainId);
```
Comment (kyzia551) [inline src/contracts/adapters/wormhole/WormholeAdapter.sol:94]: looks like it should be a part of the base adapter tbh...
Let's think about changing it in the next update (or even this)

File: src/contracts/adapters/wormhole/WormholeAdapter.sol [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+      nativeChainId,
+      REFUND_ADDRESS
+    );
+
+    return (address(WORMHOLE_RELAYER), uint256(sequence));
+  }
+
+  /// @inheritdoc IWormholeReceiver
+  function receiveWormholeMessages(
+    bytes calldata payload, // TODO: does it affect?? if in interface its as memory??
```
Comment (kyzia551) [inline src/contracts/adapters/wormhole/WormholeAdapter.sol:79]: I would keep memory as well then

**Review by kyzia551 — CHANGES_REQUESTED** [review]

File: src/contracts/adapters/wormhole/WormholeAdapter.sol (hunk 4 above)
Comment (sendra) [reply src/contracts/adapters/wormhole/WormholeAdapter.sol:79]: problem is that the subsequent call expects calldata. From the tests on testnets it works, so should we leave it as is?

File: src/contracts/adapters/wormhole/WormholeAdapter.sol (hunk 3 above)
Comment (sendra) [reply src/contracts/adapters/wormhole/WormholeAdapter.sol:94]: `_registerReceivedMessage` is an internal on baseAdapter. What could be done is move the trustedRemotes require there i think.
But provably better to do for next version, as this change will require changes on all adapters

File: src/contracts/adapters/wormhole/WormholeAdapter.sol (hunk 4 above)
Comment (kyzia551) [reply src/contracts/adapters/wormhole/WormholeAdapter.sol:79]: ok

**Review by kyzia551 — COMMENTED** [review]
Looks good, waiting for an actual params

**Comment by sendra** [comment]
Closing, as this was added on new release

**Pull request closed without merging**